extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +152,215 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * If iframe is specified in config/transport, check whether third-party
+   * iframe already exists, and if not, create it.
+   * @param {!Window} win The window element
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   * @param {function(!string,
+   *   !../../../src/3p-analytics-common.AmpAnalytics3pResponse)=}
+   *   opt_processResponse An optional function to receive any response
+   *   messages back from the cross-domain iframe
+   */
+  processCrossDomainIframe(win, transportOptions, opt_processResponse) {
+    const frameUrl = dev().assertString(transportOptions['iframe'],
+      'Cross-domain frame parameters missing ${this.type_}');
+    const frameData = this.useExistingOrCreateCrossDomainIframe_(
+      win, frameUrl, transportOptions['extraData']);
+    const iframeMessagingClient = frameData.iframeMessagingClient;
+    iframeMessagingClient.registerCallback(
+      AMP_ANALYTICS_3P_MESSAGE_TYPE.READY, () => {
+        iframeMessagingClient.setHostWindow(frameData.frame.contentWindow);
+        frameData.newCreativeMessageQueue.setIsReady();
```

Comment:
Are we positive the browser will enforce ordering here such that we can guarantee the new creative message will arrive and be processed before any events?  Otherwise events will be discarded.